synapse/handlers/auth.py

```diff
@@ -1330,7 +1331,7 @@ async def _validate_userid_login(
 
         # We raise a 403 here, but note that if we're doing user-interactive
         # login, it turns all LoginErrors into a 401 anyway.
```

Comment:
I didn't spot this comment before. Is it relevant here? Does it mean that an attacker can still infer if an account already exists?